uiux/challenge-1/src/App.js

```diff
@@ -0,0 +1,217 @@
+
+    // initialize states
+    this.state = {
+      artistName: '',
+      userStep: 'search',
+      artistInfo: [],
+      songInfo: [],
+      song: []
+    }
+  }
+
+  // enter key will initiate search
+  keyDownFunction(e) {
+    if (e.charCode === ENTER_CHAR_CODE) {
+      this.searchForArtist();
+    }
+  }
+  
+  handleClick() {
+    this.searchForArtist();
+  }
+
+  handleArtistNameChange(e) {
+    this.setState({ artistName: e.target.value });
+  }
+
+  // fetch artist data and update state
+  searchForArtist() {
+    fetch(`${BASE_URL}artist.search?q_artist=${this.state.artistName}&format=json&apikey=${API_KEY}`)
+      .then(response => response.json())
+      .then(data => {
+        this.setState({ artistInfo: data.message.body.artist_list });
+        this.setState({ userStep: 'results' });
+      });
+  }
+
+  // fetch track data and update state, filter out tracks with no lyrics as they have no use in a karaoke app
+  searchForTracks(artistId) {
+    fetch(`${BASE_URL}track.search?q_artist=${artistId}&format=json&apikey=${API_KEY}`)
+      .then(response => response.json())
+      .then(data => {
+        this.setState({ songInfo: data.message.body.track_list });
+        var newData = data.message.body.track_list.filter( (t) => {
+          if (!t.track.has_lyrics)
+            return false;
+          else
+            return t;
+        });
+
+        this.setState({ songInfo: newData });
+        this.setState({ userStep: 'songs' });
+      });
+  }
+
+  // fetch lyric data and update state
+  getLyrics(trackId) {
+    fetch(`${BASE_URL}track.lyrics.get?track_id=${trackId}&format=json&apikey=${API_KEY}`)
+      .then(response => response.json())
+      .then(data => {
+        this.setState({ song: data.message.body.lyrics });
+        this.setState({ userStep: 'playsong' });
+      });
+  }
+
+  render() {
+    // table column headers
+    const columns = [
+      {
+        Header: 'Track id',
+        show: false,
+        accessor: 'track.track_id'
+      },
+      {
+        Header: 'Track title',
+        accessor: 'track.track_name',
+        Cell: props => <a onClick={ () => this.getLyrics(props.original.track.track_id) }>{ props.value }</a>
+      },
+      {
+        Header: 'Duration (second)',
+        accessor: 'track.track_length',
+        // format track time from seconds
+        Cell: props => {
+          var track_length_seconds = props.original.track.track_length;
+          var seconds = track_length_seconds % 60;
+          var minutes = Math.floor(track_length_seconds / 60);
+
+          if (minutes < 10)
+            minutes="0"+minutes
+
+          if (seconds < 10)
+            seconds="0"+seconds
+
+          return `${minutes}:${seconds}`;
+        }
+      }
+    ]
+
+    // css
+    var fullScreen = {
+      height: '100vh',
+      background: 'linear-gradient(to right, #1a2a6c, #b21f1f, #fdbb2d)'
+    }
+
+    var font = {
+      color:'orange'
+    }
+
+    return (
```

Comment:
App.js should be importing components and render them. Why is rendering being done directly inside App.js?